### PR TITLE
Add outliers

### PR DESCRIPTION
**Pull request by MatasAdmoneo**
Implemented an algorithm that detects the mild and extreme outliers.
The algorithm returns the index of the cell in which the outlier was found.
The entire row for that index is removed.

Added some code to easier test the results. On my computer it works very poorly so extra testing would be appreciated.

### CODE REVIEW

File: dataPreProcessing.py [hunk 1]
```diff
@@ -201,4 +205,55 @@
 data.drop(rows_to_remove, inplace=True)
 
 # Reset the index after removing rows
-data.reset_index(drop=True, inplace=True)
\ No newline at end of file
+data.reset_index(drop=True, inplace=True)
+
+# Part 5 Outliers
+
+def remove_ouliers(data):
```
Comment (Ech0ke) [inline dataPreProcessing.py]: remove_outliers* (spelling)

File: dataPreProcessing.py [hunk 2]
```diff
@@ -201,4 +205,55 @@
...
+    
+    return cleaned_data
+
+# To test before and after
+# fig = px.box(data, y='Revenue')
+# fig.show()
+
+data = remove_ouliers(data)
+
+# fig = px.box(data, y='Revenue')
```
Comment (Ech0ke) [inline dataPreProcessing.py:258]: somehow we need to output side by side comparison of before and after of every column, so we don't need to switch between tabs.

File: dataPreProcessing.py [hunk 3]
```diff
@@ -201,4 +205,55 @@
...
+        # Reset the index after removing rows
+        cleaned_data.reset_index(drop=True, inplace=True)
+    
+    return cleaned_data
+
+# To test before and after
+# fig = px.box(data, y='Revenue')
+# fig.show()
+
+data = remove_ouliers(data)
```
Comment (Ech0ke) [inline dataPreProcessing.py]: save outliers into separate (e.g. ) value (e.g. no_outliers_data) not to override legacy data variable that stores initial table data.

**Review by Ech0ke — CHANGES_REQUESTED** [review]

File: dataPreProcessing.py (hunk 2 above)
Comment (MatasAdmoneo) [reply dataPreProcessing.py:258]: I do agree with your suggestion but this func is only used for testing and real visualization will be done later

**Review by Ech0ke — APPROVED** [review]

**Pull request merged**